### ISSUE TITLE
[Dev] messenger.SL3U.findAssociatedDraft() really needed?

### ISSUE THREAD

**Issue by jobisoft** [Extended-Thunder/send-later#553]
The `ComposeDetails` have a `relatedMessageId`, which is the id of the opened draft. Does that not provide the needed info?

**Comment by jph-sendlater** [comment]
You should double check, but IIRC I attempted to remove this, but had problems similar to what I mentioned in #535, where the message ID's seem to be generally unstable.

**Comment by jikamens** [comment]
I think in order for it to be safe to do this we're going to have to collect some anonymous analytics to determine if the API functionality works correctly and reliably. I've started a discussion on the mailing list about how to go about doing that:

https://thunderbird.topicbox.com/groups/addons/T3676f25e6642c8e8-Ma7852bbf6d2bc3fa6fb2264c/collecting-anonymous-analytics-from-my-add-on-to-help-improve-it

**Comment by jikamens** [comment]
In just a few hours of my telemetry change being live, I can see hundreds of instances of the related message ID in the compose details not matching the draft ID identified with the findAsssociatedDraft function in my experiement. I don't know yet why there is a difference—I haven't been able to look into that yet—but until we can determine the explanation for that it's impossible for me to switch my add-on to use relatedMessageId.

**Comment by jobisoft** [comment]
Interesting, using telemetry for this is pretty cool. I will take a closer look at what is happening there.

**Comment by jobisoft** [comment]
Ok, those are different. The draftId (used by your Experiment) refers to the opened draft, the relatedMessageId refers to the original message.

If you re-save a draft, close the composer, and open the newly saved draft, the draftId has changed, the originalMessageId does not. I have to figure out, what that means.

**Comment by jobisoft** [comment]
The original message/draft (which is `gMsgCompose.originalMsgURI`) of course no longer exists, after a re-save, the msgHdr is still lurking around and it is still retrievable with

https://searchfox.org/comm-central/rev/2fe8694a034fc959c6fd761b5b8f22f368d5463f/mail/components/extensions/parent/ext-compose.js#348-357


This will stop working after a restart, since the original/removed msgHdr is no longer retrievable.

I guess the API should return the draftId, if one is present (I checked: for a simple reply, the draftId is `""`). Do you agree?

**Comment by jikamens** [comment]
@jobisoft yes that makes sense. I'm going to remove the telemetry now because it sounds like we don't need that data anymore to identify the problem or solution.
